### ISSUE TITLE
Clarify that DMs can be read by instance owners (again)

### ISSUE THREAD

**Issue by n0toose** [mastodon/mastodon#21298]
### Pitch

Explicitly state the technical possibility of admins actually reading (or sharing) content in DMs **OR** something like that, but in a way that says "Conduct yourself like you would in public" (as in, everything you say can be made public).

Additional note: Unless if it makes sense syntactically, don't remove any of the text that exists now. (So that we all don't end up moving backwards).

### Motivation

This is a follow-up issue to https://github.com/mastodon/mastodon/issues/18079.

Even though I [previously believed that the current warning was very nice](https://github.com/mastodon/mastodon/pull/18146#issuecomment-1113510507) and also agreed with previous stances, such as the possibility that people were overreacting [over the prospect that admins would really not have any better things to do other than looking at your DMs](https://github.com/mastodon/mastodon/pull/7420#pullrequestreview-120703069).

However, recently, I read [this tweet](nitter.it/wbm312/status/1594583838201909249), as well as a tweet criticizing the entire project for not reminding instance owners as far as their legal obligations are concerned.

With all due respect to some of my favorite year-long admins and many others that will be reading this, I find it safe to conclude that this project:
- has grown exponentially; new instances are popping up and are sometimes managed by people that do not necessarily have a lot of experience with community management, which has led to drama.
- could suffer from malicious actors who can make it all worse, PR-wise — even if Mastodon is not directly liable for the conduct of everyone who chooses to host their own instance.

I believe that the following should be acknowledged:
- No matter how awesome your admin may be, sometimes, external factors that do not directly concern the morality of your instance's admins can also come into play. (including: hosting provider, admins of other instances, ability to handle both valid and invalid law enforcement requests)
- Members of the network can't always get *it* right. And that's fine, but when the *it* refers to user safety (and the safety of members of marginalized groups), proactivity is paramount. (Are there many instance owners that have money for lawyers?)
- the project should try to assist aspiring great instance owners by telling them how to do that in a way that isn't restricted to technical considerations, but that's an issue for [mastodon/documentation](https://github.com/mastodon/documentation) anyways.

A project that considers privacy so heavily (to the point where the sentence "[...] your data and your time are yours and yours alone." is included on the primary website) should always continue to find new ways in improving user agency. Honesty and setting reasonable expectations (in the place of scandalous headlines and a persistent sense of anxiety) is one of the reasons why Mastodon has won and will continue winning people over.

**Comment by n0toose** [comment]
Apologies for the long text, I felt like writing an entire manifesto (with good formatting) was the best way to highlight the urgency regarding this issue and initiate a productive conversation.

(P.S. I don't fully agree with the "Twitter did this better" part and other assessments in the linked thread, but it raises some valid points that have been said before in a sense that is actually productive.)

**Comment by phoe** [comment]
> but in a way that says "Conduct yourself like you would in public" (as in, everything you say can be made public)

I think a comment about conduct is off-topic because Mastodon, as a piece of software, has no legitimacy to tell people how to behave. That's the job of people, including moderators and administrators.

I'd instead propose a variation of the current warning which states only the dry technological facts: 

> "Posts on Mastodon are not end-to-end encrypted. Instance administrators have access to your direct messages."

I think it's enough for everyone to draw the implications themselves and, if necessary, switch to an encrypted communication tool.

**Comment by drew3000** [comment]
Agree with this approach as it heads in the direction of avoiding ambiguity on the subject.

**Comment by drew3000** [comment]
I'll even go further and suggest that the DM message button should not contain a padlock icon. This is more often associated with encrypted messaging, and gives people a false sense of security. The button could just say "direct Message" but unless it is an encrypted message the padlock should not be used.

**Comment by Cassolotl** [comment]
> However, recently, I read [this tweet](https://github.com/mastodon/mastodon/issues/nitter.it/wbm312/status/1594583838201909249)

I get a 404 when I click this. I think this is the right URL: https://twitter.com/wbm312/status/1594583838201909249

**Comment by n0toose** [comment]
Fixed, thanks @Cassolotl. I edited my original post accordingly and used `nitter.it` for privacy reasons. Self-marking this reply as off-topic.

**Comment by AlanBell** [comment]
Think it is possible to go a bit over the top on this issue. There isn't a back end user interface for admins to see reduced visibility messages. They are however in the database and not encrypted at rest - which isn't claimed and really shouldn't be expected - they are not encrypted on alternative microblogging websites. I don't think we want to imply that admins have some kind of DM browsing tool built into the system. It just isn't there, and those on managed hosting don't always have direct database access at all.

I think it is fair to say that the proprietary microblogging website has a direct messaging system that is both weirdly inconsistent with it's core public post functionality, and is also not encrypted.

**Comment by n0toose** [comment]
> There isn't a back end user interface for admins to see reduced visibility messages.

> I don't think we want to imply that admins have some kind of DM browsing tool built into the system. It just isn't there, and those on managed hosting don't always have direct database access at all.

In Mastodon. Officially. Other ActivityPub implementations may allow and support that. Legal obligations may require an administrator to look into this one way or another, as mentioned earlier. 

> those on managed hosting don't always have direct database access at all.

I included hosting in my original example, they are just an extra factor that makes this even more of a reason to be cautious instead of thinking that it's just your instance admin that defines whether your direct messages will stay private.
 
> they are not encrypted on alternative microblogging websites

I also included the example of "managed hosting" (or not) in my original post. "Twitter does it worse, so it's fine if we repeat the same mistakes" is not an argument, as long as Mastodon does not give off the impression that it is accidentally *less* secure than Twitter, which may not even be possible at this point. But come on, if someone gets curious/worried enough about this and expresses that, someone will tell them that.

Again, I am aware of what you are saying, but as I said in an excruciating amount of detail, reality can get way more complex than *"good admin, bad admin"*, but not doing anything as to not indirectly cause offense to *your* exceptional, favorite instance's admins is not a good idea -- I get that not being careful about this may fuel some distrust, healthy or not, *towards* them -- but I am going to be making it very clear once again that the underlying issues I am trying to highlight here has absolutely *nothing* to do with them and their moral integrity. That was heavily discussed in the previous iteration of the issue.

**Comment by zseri** [comment]
I think a primary aspect/focus of this (in contrast to the warning message when composing DMs) should be general improved documentation around this, also especially around legal obligations for instance admins (e.g. hosting providers, law enforcement, etc. can usually (theoretically partially, for practical reasons often also complete dumps and such) claim/demand access to data when some kind of abuse (in a broad sense of that) is suspected)

**Comment by ChanceNCounter** [comment]
Anything short of a plain-language explanation is insufficient for a lay user, and will inevitably result in another one of these down the road. What @phoe wrote above is about right. I’d reduce it to a child’s reading level. Make it literally impossible to misunderstand.

“Your messages are not secure against eavesdropping. Admins can read your DMs.”

Don’t mention e2ee or instancing, or anything else that a brand new, braindead, bottom of the barrel user might not know about. As soon as they see something that looks to them like jargon, they tune out.

**Comment by phoe** [comment]
> I don't think we want to imply that admins have some kind of DM browsing tool built into the system.

@AlanBell I think it would be a major mistake to consciously omit this possibility.

Any AP-compatible instance that Mastodon federates with to is free to add a DM-browsing functionality and [such is already present on the network](https://eldritch.cafe/@penny/109383174980439251). The fact that this software isn't Mastodon does not mean a Mastodon fork, or even an external piece of software operating on the same DB as a vanilla Mastodon instance, cannot do the same.

Implying that Mastodon DMs are secure against eavesdropping is a lie. Users should know this before sending any DM; if anything, they should be able to permanently dismiss the warning dialog in question in the Mastodon settings after clicking the equivalent to "yes, I know the security implications".

**Comment by furysama** [comment]
I think the clarification needs to clarify that it isn't just *your* admin who can read the DMs, but also the recipient's admins.  This nuance may be lost on many who aren't inherently familiar with ideas about federation, and right now that trust is very implied in the warning.

**Comment by n0toose** [comment]
You folks are seriously missing the nuance here, let me make some specific questions about the thread model that should be considered: 
- What if you go to a peaceful climate protest in a country where such assemblies can cause you to be subject to persecution, and then that said country's responsible authorities makes a formal request to your data? What if you're a journalist (happened on Twitter), a political candidate (happened on Twitter and elsewhere), an activist (happened on Twitter) -- even in places other than NA and the EU?
- Many people use Hetzner, managed hosting, and other hosting entities that somewhat resemble the faceless-ness of Twitter and supercedes whether you trust your admin or not. Jurisdiction is also affected and some resources for other decentralized services mention that (https://joinmatrix.org/servers/), I will open an issue on the website's repository in a sec.

Again, I will most likely stop being annoying if this boils down to "We should talk about the other instance's admins" as its *another* step towards the right direction.

**Comment by drew3000** [comment]
> The 🔒 icon is used for followers-only toots. The @ symbol is what is used in the UI for "Mentioned users only" toots.

Yes, that's true. But padlock icons are widely across the web to indicate security, and this instance the DMs are direct, but there is no security. Iconography gains meaning through normalised use across platforms. Calling it Direct Message or DM without the padlock does not alter the meaning, but does remove the false connotation of security or cryptographic locking.

**Comment by zseri** [comment]
@drew3000 it's even worse than that because the padlock does indicate security, but follower-only "marker" stuff appears to be basically "advisory" level stuff for remote instances (I hope tho that instances from which no one follows a person also can't access such follower-only posts, at least when Secure Fetch is enabled on the server). Maybe a "circle" symbol would be more appropriate (somewhat distorting and reusing that thematic connection from Google+)
